### PR TITLE
Update how-to-spawn-a-child-process.md

### PR DESCRIPTION
**Pull request by Yash-Handa**
Hey @chowdhurian and @fhemberger,
In accordance with issue #1977, I have updated the Article **how-to-spawn-a-child-process** with the following changes:

- Added more details about `child_process,exec()`.
- Added briefs about other spawning functions.
- some other miscellaneous updates and refactors.

If there are any **suggestion/enhancement/tips** please feel free to tell me, I would love to incorporate those changes in this Article 😄

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: locale/en/knowledge/child-processes/how-to-spawn-a-child-process.md [hunk 1]
```diff
@@ -11,29 +11,49 @@ layout: knowledge-post.hbs
...
+  console.log('Child Process STDOUT: '+stdout);
+  console.log('Child Process STDERR: '+stderr);
+});
+
+ls.on('exit', function (code) {
+  console.log('Child process exited with exit code '+code);
+});
+```
+
+NODE PRO TIP: `error.stack` is a stack trace to the point that the [Error object](https://github.com/nodejs/nodejs.org/blob/master/locale/en/knowledge/errors/what-is-the-error-object.md) was created.
```
Comment (fhemberger) [inline locale/en/knowledge/child-processes/how-to-spawn-a-child-process.md]: Link should be `/en/knowledge/errors/what-is-the-error-object/`.

File: locale/en/knowledge/child-processes/how-to-spawn-a-child-process.md [hunk 2]
```diff
@@ -11,29 +11,49 @@ layout: knowledge-post.hbs
...
 
-While `child_process.exec` buffers the output of the child process for you, it also returns a `ChildProcess` object, Node's way of wrapping a still-running process.  In the example above, since we are using `ls`, a program that will exit immediately regardless, the only part of the `ChildProcess` object worth worrying about is the `on exit` handler.  It is not necessary here - the process will still exit and the error code will still be shown on errors.
+While `child_process.exec` buffers the output of the child process for you, it also returns a `ChildProcess` object, Node's way of wrapping a still-running process. In the example above, since we are using `ls`, a program that will exit immediately regardless, the only part of the `ChildProcess` object worth worrying about is the `on exit` handler.  It is not necessary here - the process will still exit and the error code will still be shown on errors.
+
+**Buffering the Output** means that the output of the command is loaded into the memory before sending to `STDOUT` or `STDERR` and as mentioned above a default of 200KB can be buffered into the memory. This feature has both pros and cons:
```
Comment (fhemberger) [inline locale/en/knowledge/child-processes/how-to-spawn-a-child-process.md]: Should be lowercase "`stdout` or `stderr`"

**Review by fhemberger — CHANGES_REQUESTED** [review]
LGTM, two little nits.

File: locale/en/knowledge/child-processes/how-to-spawn-a-child-process.md (hunk 1 above)
Comment (fhemberger) [reply locale/en/knowledge/child-processes/how-to-spawn-a-child-process.md]: https://nodejs.org/en/knowledge/errors/what-is-the-error-object/ works, please revert your last commit to `/en/knowledge/errors/what-is-the-error-object/`. 😉

**Review by ghost — APPROVED** [review]

**Review by fhemberger — APPROVED** [review]

**Pull request merged**